README.md

```diff
@@ -1 +1,68 @@
+valid data is received, the callback passed to setReceiveHandler will be called with the received data as its argument
+
+## readReg(address)
+The readReg(address) method returns the value stored at the register addressed by the passed address parameter
+
+## writeReg(address, data)
+The writeReg(address, data) method writes the data parameter to the register addressed by the address parameter
+
+## setReceiveHandler(handler)
+The setReceiveHandler(handler) method takes a function as its parameter. This function should be prepared to take a single
+string parameter as its input. It will be called whenever data is received, with the received data passed to it as its parameter
+
+## maskAllInterrupts() 
+The maskAllInterrupts() method will mask all interrupts.
+
+## enableInterrupt(mask)
+The enableInterrupt(mask) method will enable the interrupt corresponding to the mask parameter. The mask parameter should be
+one of the class constants corresponding to interrupts. These are RX_TIMEOUT, RX_DONE, PAYLOAD_CRC_ERROR, VALID_HEADER, TX_DONE,
+CAD_DONE, FHSS_CHANGE_CHANNEL, and CAD_DETECTED.
+
+## isDoneSending()
+The isDoneSending() method will return whether the last message to be transmitted has finished sending
+
+## setters
+The following methods are available to set various radio parameters:
+- setPayloadLength(len): this method takes a length 0-255 as a parameter, and should only be necessary in implicit header mode
+- setImplicitHeaderMode(state): this method takes a boolean indicating whether to operate in implicit header mode
+- setPreambleLength(len): this method takes a two byte parameter for the length of the preamble
+- setFifoTxBase(start): this method sets the start of Tx data in the FIFO buffer
+- setFifoRxBase(start): this method sets the start of Rx data in the FIFO buffer
+- setCodingRate(cr): this method sets the coding rate. It takes a parameter which is mapped via a table. The available values are 4/5, 4/6, 4/7,
+and 4/8. These should be passed as strings like so: "4/5"
+- setRxPayloadCRC(state): this method takes a boolean indicating whether received packet headers indicate CRC being on
+- setBandwidth(bw): this method takes sets the bandwidth for the transceiver. It takes a paramter which is mapped via a table.
+The available values in kHz are 7.8, 10.4, 15.6, 20.8, 31.25, 41.7, 62.5, 125, 250, 500. These should be passed as strings like
+so: "125"
+- setMode(mode): this method sets the mode of the chip. The available modes, which are class constants, are SLEEP, STANDBY, FSTX,
+TX, FSRX, RXCONTINUOUS, RXSINGLE, and CAD. They should be passed to setMode as its sole parameter
+- setSpreadingFactor(sf): this method sets the spreading factor of the chip. Only 6-12 can be passed as parameters
+- setMaxPayload(pl): this method sets the maximum payload size.
+- setFrequency(freq): this method sets the desired frequency for the chip. See the datasheet for specifics on which frequencies
+can be used
+
```

Comment:
Please give a link to the license file, like at https://github.com/electricimp/messagemanager